backend/pkg/kubeconfig/kubeconfig.go

```diff
@@ -281,334 +304,412 @@ func LoadContextsFromBase64String(kubeConfig string, source int) ([]Context, []e
-	rawContexts, _ := rawConfig["contexts"].([]interface{})
-	rawClusters, _ := rawConfig["clusters"].([]interface{})
-	rawUsers, _ := rawConfig["users"].([]interface{})
+	// Get the contexts from the kubeconfig
+	rawContexts, err := GetContextsFromKubeconfig(kubeconfig)
+	if err != nil {
+		return nil, []error{err}
+	}
 
-	// Create contexts from the parsed data.
+	// Process each context
 	for _, rawContext := range rawContexts {
-		context, err := createContext(rawContext, rawClusters, rawUsers, source)
+		context, err := ProcessContext(rawContext, kubeconfig, source, skipProxySetup)
 		if err != nil {
 			errs = append(errs, err)
-			continue
 		}
 
 		contexts = append(contexts, context)
 	}
 
-	// Return the contexts and any errors that occurred.
 	return contexts, errs
 }
 
-// parseAndValidateConfig parses and validates a kubeconfig file.
-// It returns the raw config, the parsed config, and any errors that occurred.
-// It returns an error if the file cannot be parsed.
-// It returns a valid config if the file is parsed and validated.
-// It returns any errors that occurred during parsing or validation.
-func parseAndValidateConfig(data []byte) (map[string]interface{}, *api.Config, error) {
-	var rawConfig map[string]interface{}
+// UnmarshalKubeconfig unmarshals the kubeconfig data.
+func UnmarshalKubeconfig(data []byte) (map[string]interface{}, error) {
+	var kubeconfig map[string]interface{}
 
-	err := yaml.Unmarshal(data, &rawConfig)
+	err := yaml.Unmarshal(data, &kubeconfig)
 	if err != nil {
-		return nil, nil, err
+		return nil, DataError{Field: "kubeconfig", Reason: fmt.Sprintf("error unmarshaling YAML: %v", err)}
 	}
 
-	config, err := clientcmd.Load(data)
-	if err != nil {
-		return rawConfig, nil, err
-	}
+	return kubeconfig, nil
+}
 
-	if err := clientcmd.Validate(*config); err != nil {
-		return rawConfig, config, err
+// GetContextsFromKubeconfig gets the contexts from the kubeconfig.
+func GetContextsFromKubeconfig(kubeconfig map[string]interface{}) ([]interface{}, error) {
+	rawContexts, ok := kubeconfig["contexts"].([]interface{})
+	if !ok {
+		return nil, DataError{Field: "contexts", Reason: "invalid or missing contexts in kubeconfig"}
 	}
 
-	return rawConfig, config, nil
+	return rawContexts, nil
 }
 
-// createContext creates a context from the given raw context, clusters, and users.
-// It returns the created context and any errors that occurred.
-// It returns an error if the context cannot be created.
-// It returns a valid context if the context is created.
-// It returns any errors that occurred during context creation.
-func createContext(rawContext interface{}, rawClusters, rawUsers []interface{}, source int) (Context, error) {
-	contextMap, ok := rawContext.(map[interface{}]interface{})
-	if !ok {
-		return Context{}, ContextError{ContextName: "unknown", Reason: "invalid context format"}
-	}
+// ProcessContext processes a context from the kubeconfig.
```

Comment:
I'm not sure what some of these arguments are. Can you please document them along with the return?